### PR TITLE
[ENHANCEMENT] - Add string shorten function

### PR DESCRIPTION
**Pull request by StyxOfDynamite**
## Description
New snippet for returning a string shortened to a given length. with `...` appended by default.

## PR Type
- [x] Snippets, Tests & Tags (new snippets, updated snippets, re-tagging of snippets, added/updated tests)
- [ ] Tools, Scripts & Automation (anything related to files in the scripts folder, Gatsby, website, Travis CI or Netlify)
- [ ] General, Typos, Misc. & Meta (everything related to content, typos, general stuff and meta files in the repository - e.g. the issue template)
- [ ] Other (please specifiy in the description above)

## Guidelines
- [x] I have read the guidelines in the [CONTRIBUTING](https://github.com/30-seconds/30-seconds-starter/blob/master/CONTRIBUTING.md) document.

### CODE REVIEW

**Comment by peter279k** [comment]
Hi @StyxOfDynamite, just notify you that this PR will not be merged temporarily.

I should resolve issue #69 firstly and rebuilding the test system.

**Comment by peter279k** [comment]
BTW, I wonder why the code snippets building process is failed :thinking:...

@Chalarangelo, do you have any idea about this?

File: snippets/shorten.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title:  shorten
```
Comment (Chalarangelo) [inline snippets/shorten.md]: Remove the additional space before the title here.

File: snippets/shorten.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+tags: string,beginner
+---
+
+Returns a shortened string
```
Comment (Chalarangelo) [inline snippets/shorten.md]: Add a `.` at the end.

File: snippets/shorten.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+Use `mb_strlen()`, `mb_substr()` and `rtrim()` to shorten a string to a give number of characters.
+
+```php
+function shorten($input, $length = 100, $end = '...')
+{
```
Comment (Chalarangelo) [inline snippets/shorten.md]: Indentation should be 2 spaces per level, please fix for the whole snippet.

File: snippets/shorten.md (hunk 1 above)
Comment (StyxOfDynamite) [reply snippets/shorten.md]: Completed.

File: snippets/shorten.md (hunk 2 above)
Comment (StyxOfDynamite) [reply snippets/shorten.md]: Completed.

File: snippets/shorten.md (hunk 3 above)
Comment (StyxOfDynamite) [reply snippets/shorten.md]: Completed. (Had some issues getting .git to notice the changes but they should be there.

**Comment by StyxOfDynamite** [comment]
Apologies for the overhead with this one.

**Comment by Chalarangelo** [comment]
@peter279k it might be Netlify acting up or something, I am running a fresh deploy to see what's going on.

**Comment by Chalarangelo** [comment]
Still failing, but it fails only for this snippet, I will test manually and deploy if it works, which it should btw.

**Comment by Chalarangelo** [comment]
Nevermind, it's waiting in a Travis-generated file, should be file when merged.

**Comment by StyxOfDynamite** [comment]
Thanks all, feels good to get this PR accepted :)

**Pull request merged**